http/src/main/java/io/micronaut/http/body/DynamicMessageBodyWriter.java

```diff
@@ -49,22 +48,13 @@ public MessageBodyWriter<Object> createSpecific(Argument<Object> type) {
         return registry.findWriter(type, mediaTypes).orElse(this);
     }
 
-    public MessageBodyWriter<Object> find(Argument<Object> type, MediaType mediaType, Object object) {
-        MessageBodyWriter<Object> specific = registry.findWriter(type, List.of(mediaType)).orElse(null);
-        if (specific != null && !(specific instanceof DynamicMessageBodyWriter)) {
-            return specific;
+    public <T> MessageBodyWriter<T> find(Argument<T> type, MediaType mediaType, T object) {
+        if (type.getType() == Object.class) {
+            type = Argument.of(object);
```

Comment:
if you have this logic you don't need to change all the callers. just keep them as OBJECT_ARGUMENT

Reply:
But that's the call to the registry not to the dynamic writer. The dynamic writer will skip dynamic writers

Reply:
Then writeTo should also be adjusted. The whole point of DynamicMessageBodyWriter is to work with OBJECT_ARGUMENT

Reply:
Eliminated `DynamicMessageBodyWriter` completely